### ISSUE TITLE
Thermos 55 + Multiverse 2.5 causes client crash on relog

### ISSUE THREAD

**Issue by Brycey92** [TCPR/Thermos#313]
**Server Log:**
https://gist.github.com/Brycey92/59a4f5bc6e7a2c1e906a1d0a6e64d344

**FML Log(s):**
https://gist.github.com/Brycey92/02ad89a21b775333a95e4cd4bc8355ff

**Explanation of issue:**
Upon joining my server running Thermos 55 and Multiverse 2.5-b719, everything appears normal. Upon leaving and rejoining any time after that, large amounts of blocks appear client-side replacing both existing blocks and air blocks. The client fps drops accordingly until it crashes for attempting to display some block with out-of-range metadata 8. In this log, it was Natura, but it was Galacticraft in a different log, for example.

I have not tested if this issue occurs with Multiverse 2.4, but it would be ideal if a fix could be implemented either in Multiverse or Thermos for Multiverse 2.5.

**How to recreate this issue:**
Have Thermos 55, MV 2.5-b719
Start server
Join server
Leave server
Join server again
Crash or disconnect if you can in time
Profit???

**Modpack Name:** *(Only if public)*
Galaxy-Craft Modpack

**Mods Installed:**
http://www.technicpack.net/modpack/galaxy-craft-modpack.210706/mods

**Plugins Installed:**
AsyncWorldEditInjector, Multiverse-Core, Multiverse-Portals, WorldEdit, AsyncWorldEdit, OpenInv, Essentials, PermissionsEx, BukkitCompat, PlugMan, WorldBorder, Vault, EssentialsSpawn, GriefPrevention, ItemRestrict, WorldGuard, CommandHelper, Herochat

**Warmroast Report:** *(Optional)*

**Thermos Version:**
55

**Forge Version:**
1614

**Comment by Time6628** [comment]
Something with Natura I think.

**Comment by Brycey92** [comment]
No, it's not Natura. This happened with Galacticraft too https://gist.github.com/Brycey92/4c300473f12e5a253eecc674976eea9a

**Comment by Brycey92** [comment]
The issue is not present when the Multiworld plugin in place of Multiverse.
It also is not present on any Forge 1614 KCauldron build.

**Comment by DiggyWig** [comment]
Try Multiverse-Core Version: 2.5-b691 & Multiverse-Portals Version: 2.5-b699, I had issues with higher versions but those work perfectly for me.

**Comment by djoveryde** [comment]
Yh 2.5 for me didn't work at all, I had to use 2.4 for mine to work :(

**Comment by Brycey92** [comment]
You've tried the builds DiggyWig suggested, djoveryde?

**Comment by Brycey92** [comment]
The issue is still present on those builds.

**Comment by Time6628** [comment]
Does it do it without Multiverse?

**Comment by Brycey92** [comment]
No

**Comment by Time6628** [comment]
Try removing FastCraft.

**Comment by Brycey92** [comment]
Removing FastCraft has no effect.

**Comment by Brycey92** [comment]
Another thing I noticed: in certain areas, the client will not crash, but will instead lag a ton. I found that the client, on top of rendering ghost blocks and lagging, renders the world as a Galacticraft planet, with the extra stars and earth in the sky. I will test with other versions of Galacticraft tomorrow.

**Comment by djoveryde** [comment]
@Brycey92 yup, all 2.5 versions failed for me.

**Comment by Time6628** [comment]
I believe it is something to do with FastCraft modifying block rendering, I am not sure though because FastCraft is not open source but from you CrashReport fastcraft is definatly involved, if it still crashes without fastcraft please post that crash report.

**Comment by Brycey92** [comment]
Before removing FastCraft, I had moved to a location on the server that didn't crash my game upon the rendering issues occurring, and I don't know where the original area was that caused the crash. Removing FastCraft, however, still resulted in the exact same rendering issues at the new location. If I am able to find the area where the crashes happened, I will test it again without FastCraft. For now, however, I can tell you FastCraft is not causing the rendering issues themselves, whether or not it was causing the subsequent crashes.

**Comment by Brycey92** [comment]
Strangely, the issue does not exist when Galacticraft is downgraded to build 406 instead of 454. I will continue to test other Galacticraft builds.

**Comment by Brycey92** [comment]
Sorry for all the title changes!

**Comment by Nentify** [comment]
I had similar issues recently as well! Maybe this also affect the ability to use plugin generators? I couldn't get a new void/empty world with either MV generators or MW. After trying to make this new world, I started getting these weird rendering issues and occasional crashes when joining the server.

**Comment by Brycey92** [comment]
Anyone looking for the latest working version of Galacticraft: https://ci.micdoodle8.com/job/Galacticraft-1.7/412/

**Comment by DraksterAU** [comment]
@brycey92 use build 446 of GC and it works fine

**Comment by DraksterAU** [comment]
@Time6628 I believe this is caused by Thermos, not GC as it does not occur in B46 or earlier (I haven't moved past that for this reason).

**Comment by Time6628** [comment]
Just tested this with the latest successful Galaticraft build, no issues at all. Then again I didn't test with mutliverse.

**Comment by DraksterAU** [comment]
@Time6628 Its the bukkit side of Thermos, not Multiverse. (Multiworld is broken too in later thermos builds)

**Comment by Time6628** [comment]
Which multiworld?

**Comment by DraksterAU** [comment]
@Time6628 There is only 1. Bukkit Multiworld.

The forge one is a joke compared.

**Comment by Time6628** [comment]
Does it happen without multiverse or whatever world plugin you are using?

**Comment by djoveryde** [comment]
Adding to the Issue, using Multiverse 2.4 with Galacticraft as soon as I create a Multiverse world randomly Group Manager starts moving players to the default rank every minute. Its like somethings screwing with bukkit side. No console messages at all

**Comment by Time6628** [comment]
Solution: don't use multiverse.

**Comment by DraksterAU** [comment]
@time6628 You need to make sure you continue support for at least multi world OR multiverse.... Or people won't update.

Thermos in its later versions is breaking things that previously worked (not related to this issue) that are preventing a lot of us from doing so.

Don't want to sound like a douche... But it's true :(

**Comment by djoveryde** [comment]
@Time6628 I think the idea here is to correct things that get broken by thermos, not brush them under the carpet and say "don't use multiverse/multiworld". 
Basically in essence thats the only plugins that allow for multi world creation which is used by people that want their spawn to be in its own protected world away from stuff that can break it.
The plugin itself works fine when you haven't made a world using it, but as soon as you do and you go there thats when the spam attacks start on all the players. 
The issue needs to be fixed as this is a pretty major issue.

How to replicate: Multiverse or Multiworld Plugin, Group Manager Plugin, galacticraft (not sure if this is needed or not to replicate).
2 Players needed.
Create a new world using the world plugin, go to that world. Whoever is on the normal overworld will get spammed with "You have been moved to the group Default" every minute until the other player leaves and deletes the MV World.

**Comment by Nentify** [comment]
What solution has been determined for this?

This rendering issue occurs when I leave and join the server for a second time.

Here's an error from the client: http://pastebin.com/raw/WnJg3PvP

**Comment by DraksterAU** [comment]
Use Build 46 Thermos. Seems to work with everything.

**Comment by DraksterAU** [comment]
[14:01:23] [Client thread/ERROR]: Dimension already registered to another mod: unable to register planet dimension 0
[14:01:23] [Client thread/ERROR]: Dimension already registered to another mod: unable to register planet dimension 0

Possible that's because Dim Register packets are firing in the mod and in thermos now? I know little about Java... so....

**Comment by Nentify** [comment]
I found that this happens after using /mv load DIM-28...30 (the 3 GC worlds). If I remove them from my worlds.yml file, the error doesn't occur.

**Comment by Robotia** [comment]
It might be because Thermos is considering these worlds to be Bukkit provided worlds, in which case Nentifys finding would make sense.

**Comment by Brycey92** [comment]
Would there be a way to detect or manually specify which dimensions are Forge provided and which are Bukkit provided? (Yay, necroposting!)

**Comment by Brycey92** [comment]
If you don't need any of the teleporting features of bukkit into your GC space dimensions, you can follow Nentify's comment and remove them from your GC config. This fixes the issue without getting rid of Multiverse, Thermos 47+, or GC 413+, and it seems to have the smallest impact. Other bukkit features seem to still work in space, like GriefPrevention land claiming, though I haven't done any thorough tests.
